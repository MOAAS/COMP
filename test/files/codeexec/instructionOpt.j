.class public SmallOptimizations
.super java/lang/Object
.method public static main([Ljava/lang/String;)V
.limit stack 3
.limit locals 3
new SmallOptimizations
dup
invokespecial SmallOptimizations/<init>()V
astore_2
aload_2
iconst_1
invokevirtual SmallOptimizations/double(I)I
istore_1
aload_2
bipush 10
invokevirtual SmallOptimizations/negate(I)I
invokestatic io/println(I)V
aload_2
bipush 10
invokevirtual SmallOptimizations/double(I)I
invokestatic io/println(I)V
iload_1
bipush 10
imul
invokestatic io/println(I)V
iload_1
bipush 2
ishl
invokestatic io/println(I)V
iload_1
bipush 4
ishl
iconst_1
iadd
invokestatic io/println(I)V
aload_2
bipush 30
invokevirtual SmallOptimizations/double(I)I
istore_1
iload_1
bipush 5
ishr
iload_1
bipush 6
ishl
isub
invokestatic io/println(I)V
return
.end method

.method public negate(I)I
.limit stack 1
.limit locals 2
iload_1
ineg
ireturn
.end method

.method public double(I)I
.limit stack 2
.limit locals 2
iload_1
bipush 1
ishl
ireturn
.end method

.method public <init>()V
  aload_0
  invokenonvirtual java/lang/Object/<init>()V
  return
.end method

