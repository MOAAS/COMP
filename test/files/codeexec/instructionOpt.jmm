import io;
import static io.println(int);

class SmallOptimizations {
	public static void main(String[] args) {
	    int a;
	    SmallOptimizations obj;

	    obj = new SmallOptimizations();
	    a = obj.double(1);

	    io.println(obj.negate(10));
	    io.println(obj.double(10));
	    io.println(a * 10);
	    io.println(a * 4);
	    io.println(a * 16 + 1);

	    a = obj.double(30);

	    io.println(a / 32 - a * 64);
	}


	public int negate(int num) {
	    return 0 - num;
	}

    public int double(int num) {
        return num * 2;
    }
}