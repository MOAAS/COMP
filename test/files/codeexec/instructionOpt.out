-10
20
20
8
33
-3839